src/services/identity/SitesService.ts

```diff
@@ -78,6 +84,107 @@ class SitesService {
     return user.site_members.map((site) => site.repo?.name)
   }
 
+  async getCommitAuthorEmail(commit: GitHubCommitData) {
+    const { message } = commit
+
+    // Commit message created as part of phase 2 identity
+    if (message.startsWith("{") && message.endsWith("}")) {
+      const { userId }: { userId: string } = JSON.parse(message)
+      const user = await this.usersService.findById(userId)
+
+      if (user) {
+        return user.email
+      }
+    }
+
+    // Legacy style of commits, or if the user is not found
+    const {
+      author: { email: authorEmail },
+    } = commit
+    return authorEmail
+  }
+
+  async getUrlsOfSite(
+    sessionData: UserWithSiteSessionData,
+    urlTypes: SiteUrlTypes[]
+  ): Promise<SiteUrls> {
+    // Tries to get the site urls in the following order:
+    // 1. From the deployments database table
+    // 2. From the config.yml file
+    // 3. From the GitHub repository description
+    // Otherwise, returns an empty string for the url
+    const { siteName } = sessionData
+    let configYmlData: ConfigYmlData = {}
+    let repoDescTokens: string[] = []
+
+    const site = await this.siteRepository.findOne({
+      where: { name: siteName },
+      include: {
+        model: Deployment,
+        as: "deployment",
+      },
+    })
+
+    if (!site) {
+      return {
+        staging: "",
+        prod: "",
+      }
+    }
+
+    const deployment = site?.deployment
+    const output: SiteUrls = {
+      staging: deployment?.stagingUrl ? deployment.stagingUrl : "",
+      prod: deployment?.productionUrl ? deployment.productionUrl : "",
+    }
```

Comment:
note that `Deployment` has `allowNull: false` set on both of these so we probably shouldn't do this; instead, what we could do is to define a `doesDeploymentExists` variable that determines if we should fetch using `configYmlService` (or just `!deployment`) 

this does make it slightly less future proof (eg: we want to `allowNull`) though... lmk what you think!

Reply:
If deployment (or site) does not exist, then we will just continue which is intended.